chain/blockvalidator.go

```diff
@@ -91,6 +91,18 @@ func TransactionCheck(ctx context.Context, block *block.Block) error {
 		txnsHash[i] = txn.Hash()
 	}
 
+	txnsRoot, err := crypto.ComputeRoot(txnsHash)
+	if err != nil {
+		return fmt.Errorf("compute txns root error: %v", err)
+	}
+	if !bytes.Equal(txnsRoot.ToArray(), block.Header.UnsignedHeader.TransactionsRoot) {
+		return fmt.Errorf("computed txn root %x is different from txn root in header %x", txnsRoot.ToArray(), block.Header.UnsignedHeader.TransactionsRoot)
+	}
+
+	if DefaultLedger.Store.DuringFastSync(block.Header.UnsignedHeader.Height) {
```

Comment:
Txn hash is not affected by signature (same for block hash), so signature needs to be verified as well